apps/crl.c

```diff
@@ -232,27 +233,37 @@ int crl_main(int argc, char **argv)
             goto end;
         }
 
-        xobj = X509_STORE_CTX_get_obj_by_subject(ctx, X509_LU_X509,
-                                                 X509_CRL_get_issuer(x));
-        if (xobj == NULL) {
-            BIO_printf(bio_err, "Error getting CRL issuer certificate\n");
+        matching_objs = X509_STORE_CTX_get1_objs_by_auth(ctx,
+                                                         X509_CRL_get_issuer(x),
+                                                         X509_CRL_get0_authority_key_id(x));
```

Comment:
Per [RFC5280, Section 5.2.1](https://datatracker.ietf.org/doc/html/rfc5280#section-5.2.1) the authority key id is required in CRLs, and MUST specify a key id, rather than issuer DN + serial. Therefore, it may be appropriate to check that `X509_CRL_get0_authority_key_id` returns a non-NULL value. Otherwise the CRL `x` is non-conformant, and could be rejected (fail to validate).